docs/build/build-with-ai/build-with-goat/send-token-agent.md

```diff
@@ -0,0 +1,228 @@
+
+Follow these steps to set up the AI-powered token sending agent on Celo:
+
+**1. Clone Repository and Navigate to Example Directory:**
+
+Follow steps 1-5 from the NFT minting tutorial (or previous tutorials) to clone the GOAT repository, navigate to the `typescript` directory, install dependencies, build the project, and go to the example directory: `examples/by-use-case/evm-mint-nft`. You can reuse this directory or create a new one if you prefer.
+
+**2. Configure Environment Variables:**
+
+```bash
+cp .env.template .env
+```
+
+Copy `.env.template` to `.env` and populate the following environment variables in the `.env` file:
+
+- **`OPENAI_API_KEY`**: Your OpenAI API key from [OpenAI](https://www.google.com/url?sa=E&source=gmail&q=https://platform.openai.com/).
+- **`WALLET_PRIVATE_KEY`**: Your private key for the wallet that will send tokens on Celo. **Security Best Practices:** Use a test wallet and handle private keys with extreme caution.
+- **`RPC_PROVIDER_URL`**: The RPC URL for the Celo network. Use a Celo Alfajores Testnet RPC URL for testing (e.g., `https://alfajores-forno.celo-testnet.org`). Refer to previous articles for more Celo RPC options.
+
+**Example `.env` file (for Celo Alfajores Testnet):**
+
+```
+OPENAI_API_KEY=YOUR_OPENAI_API_KEY
+WALLET_PRIVATE_KEY=YOUR_PRIVATE_KEY
+RPC_PROVIDER_URL=[https://alfajores-forno.celo-testnet.org](https://alfajores-forno.celo-testnet.org)
+```
+
+**Note:** We only need `OPENAI_API_KEY`, `WALLET_PRIVATE_KEY`, and `RPC_PROVIDER_URL` for this token sending agent.
+
+**3. Adapt Code for Celo Token Sending:**
+
+1.  **Chain Configuration:** In `index.ts`, replace `baseSepolia` with `celo` from `viem/chains`:
+
+    ```javascript
+    import { celo } from "viem/chains"; // Import Celo chain
+
+    // ...
+
+    const walletClient = createWalletClient({
+      account: account,
+      transport: http(process.env.RPC_PROVIDER_URL),
+      chain: celo, // Use Celo chain configuration
+    });
+    ```
+
+2.  **Adapt `sendETH()` to `sendCELO()`:** The `@goat-sdk/wallet-evm` might have a function specifically for sending CELO. Check the `@goat-sdk` documentation for `sendCELO()`. If it exists, replace `sendETH()` with `sendCELO()` in the `plugins` array:
+
+    ```javascript
+    plugins: [
+      sendCELO(), // Enable CELO transfers (if available in @goat-sdk)
+      erc20({ tokens: [USDC, PEPE] }), // ERC20 token operations - Review tokens for Celo
+      // ...
+    ];
+    ```
+
+    If `sendCELO()` is not directly available in `@goat-sdk/wallet-evm`, it's likely that `sendETH()` is designed to be chain-aware and will send the native token of the configured chain (`celo` in this case). In that case, you might not need to change `sendETH()`. **Test sending CELO after setup to confirm if `sendETH()` works for CELO or if you need to find an alternative.** If `sendETH()` does not work for CELO, you might need to create a custom tool using `viem`'s `sendTransaction` function to send CELO directly.
```

Comment:
Why would it not be available? is this a version issue? Can you provide the function for the workaround?